src/FormAlter/MediaFormAlter.php

```diff
@@ -98,25 +92,26 @@ public function reimportCallback(array $form, FormStateInterface $formState) {
    *   The name of the field holding the mpx ID, or NULL if not configured.
    */
   private function resolveIdFieldName(Media $video):? string {
-    if (!$field_map = $video->bundle->entity->getFieldMap()) {
-      return NULL;
-    }
-
-    return isset($field_map['Media:id']) ? $field_map['Media:id'] : NULL;
+    $media_source = $video->getSource();
```

Comment:
Turns out this field mapping is not mandatory (which was an assumption I'd made when developing all the UI work). The value from the source field definition is what we need.